src/server/replica.h

```diff
@@ -49,11 +50,11 @@ class Replica {
   // A generic barrier that is used for waiting for
   // flow fibers to become ready for the stable state switch.
   struct SyncBlock {
-    SyncBlock(unsigned flows) : flows_left{flows} {
-    }
-    unsigned flows_left;
+    unsigned flows_done{0};
     ::boost::fibers::mutex mu_;
     ::boost::fibers::condition_variable cv_;
+
+    void Add(unsigned delta);
```

Comment:
why BlockingCounter does not fit it?

Reply:
Hm... I need to change it though a little

In case of an error, I need a way to unblock it. I can add a function like `Reset()` which resets the count to `0`. However, flows still might decrement it and we'll get an underflow. 

So I either:
* Use an int inside
* Use a CAS decrement inside the blocking counter
* Allow locking the context externally, so I can do
```
with cntx.Lock()
  if not cntx.Error():
    counter.Dec()
```

What do you think?

Reply:
You can do it differently. You do not really need 64 bit for count.
You can use the high bit as cancel bit. I do not like Reset but you can use 
```cpp
Cancel()  {
 count_.fetch_or(1ULL << 63, std::memory_order_acquire);
  ec_.notify();  // releases inside
}

void Wait() {
  ec_.await([this] { 
    auto cnt =  count_.load(std::memory_order_acquire); 
     return cnt == 0 || (cnt &  (1 << 63));
  });
}
```